protocol/src/main/java/org/sdo/pri/OnDieSignatureValidator.java

```diff
@@ -0,0 +1,197 @@
+import org.bouncycastle.cert.X509CertificateHolder;
+import org.slf4j.LoggerFactory;
+
+
+public abstract class OnDieSignatureValidator {
+
+
+  /**
+   * Performs a validation of the given signature with the public key
+   * extracted from the cert chain.
+   *
+   * @param signedData signedData
+   * @param signature signature
+   * @param publicKey public key to validate signature with
+   * @return boolean indicating if signature is valid.
+   */
+  public static boolean validateWithoutRevocations(
+          String signedData,
+          byte[] signature,
+          PublicKey publicKey) {
+
+    byte[] adjSignature = new byte[0];
+    // check minimum length (taskinfo + R + S)
+    if (signature.length < (36 + 48 + 48)) {
+      return false;
+    }
+    try {
+      // first 36 bytes are always the taskInfo
+      byte[] taskInfo = Arrays.copyOfRange(signature, 0, 36);
+
+      // adjust the signed data
+      // data-to-verify format is: [ task-info | nonce (optional) | data ]
+      // First 36 bytes of signature is the taskinfo. This value must be prepended
+      // to the signedData
+      ByteArrayOutputStream adjSignedData = new ByteArrayOutputStream();
+      adjSignedData.write(taskInfo);
+      adjSignedData.write(signedData.getBytes());
+
+      adjSignature = convertSignature(signature, taskInfo);
+
+      Signature sig = Signature.getInstance("SHA384withECDSA");
+      sig.initVerify(publicKey);
+      sig.update(adjSignedData.toByteArray());
+      return sig.verify(adjSignature);
+    } catch (Exception ex) {
+      return false;
+    }
+  }
+
+  /**
+   * Performs a validation of the given signature with the public key
+   * extracted from the given cert chain.
+   *
+   * @param signedData signedData
+   * @param signature signature
+   * @param certPath certificate path (dc value from ov)
+   * @param onDieCache cache that contains the CRLs
+   * @return boolean indicating if signature is valid.
+   * @throws CertificateException when error.
+   */
+  public static boolean validateWithRevocations(
+          String signedData,
+          byte[] signature,
+          CertPath certPath,
+          OnDieCache onDieCache) throws CertificateException {
+
+    // Check revocations first.
+    List<Certificate> certificateList = (List<Certificate>) certPath.getCertificates();
+
+    // Check revocations first.
+    if (!checkRevocations(certificateList, onDieCache)) {
+      return false;
+    }
+
+    // Now validate the signature
+    // Public key comes from the cert chain in the voucher.
+    if (certificateList.size() == 0) {
+      throw new CertificateException("OnDieSignatureValidation: Certificate chain is empty.");
+    }
+    return validateWithoutRevocations(signedData, signature, certificateList.get(0).getPublicKey());
+  }
+
+
+  private static byte[] convertSignature(byte[] signature, byte[] taskInfo)
+        throws IllegalArgumentException, IOException {
+    if (taskInfo.length != 36) {
```

Comment:
It would be helpful for better readability to define the variables/constants for the hard-coded numbers (36/48/49/96) and put some comments on what they mean.

Reply:
done